Vagrantfile

```diff
@@ -0,0 +1,35 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+# Author: David Manouchehri
+
+Vagrant.configure("2") do |config|
+	config.vm.box = "bento/ubuntu-16.04"
+
+	config.vm.provision "shell", inline: <<-SHELL
+		# apt-get update
+		# DEBIAN_FRONTEND=noninteractive apt-get -y upgrade
+		snap install --classic go
+		# apt-get clean
+	SHELL
+
+	config.vm.provision "docker"
+
+	config.vm.provision "shell", privileged: false, inline: <<-SHELL
+		mkdir -p ~/go/src/github.com/linuxkit/
+		ln -s /vagrant ~/go/src/github.com/linuxkit/linuxkit
+		cd ~/go/src/github.com/linuxkit/linuxkit/src/cmd/linuxkit
+		go get -d
+		go install
```

Comment:
If you use `make install` instead of these two then the binaries will be correctly tagged with the git commit sha and version. You can override `PREFIX` if you don't want (or can't) install into `/usr/local/bin`. Using `PREFIX=$HOME/go` would  be compatible with the following.

Reply:
Okay, if I do this, then we won't be reusing the existing cloned repository like you asked me in https://github.com/linuxkit/linuxkit/pull/2784#pullrequestreview-80121346. Looks like in https://github.com/linuxkit/linuxkit/blob/13adecbc3e136fcb1894f0604db04468dd01a4b3/Makefile#L81 the repository is being cloned again.

I don't mind doing either, but spending time testing changes only to have them reverted in a couple hours is a bit frustrating. (Rebuilding the entire VM takes a few minutes, regardless of how many lines are changed. It's not like Docker where every RUN layer is cached.)

Reply:
`make install`'s reclone of the repo to build the tool is orthogonal to the desire to leave the user with a local working tree (where they can edit `pkg/*`, build packages etc which I think is the point of this Vagrant file, isn't it?) which is not a duplicate of the one of the host.

So please both reuse the host repo (as requested in previous review) and use `make install`, there is no contradiction there AFAICT, sorry if it appeared that way.

Reply:
I replaced `go get -d` and `go install` with `make install` like you asked, and it doesn't work.

```
mkdir -p ~/go/src/github.com/linuxkit/
ln -s /vagrant ~/go/src/github.com/linuxkit/linuxkit
cd ~/go/src/github.com/linuxkit/linuxkit
make install
echo "export PATH=${PATH}:${HOME}/go/bin" >> ~/.bashrc
```

```bash
vagrant@vagrant:~/go/src/github.com/linuxkit/linuxkit$ make install
cp -R ./bin/* /usr/local//bin
cp: cannot stat './bin/*': No such file or directory
Makefile:92: recipe for target 'install' failed
make: *** [install] Error 1
```

Reply:
Weird, looks like `install` misses the dependency to actually build the binaries first. You should insert a `make all` in front of the `make install` (or you could look into fixing up the `Makefile`).

Reply:
Okay, done.

When possible, could you test changes locally before requesting them?

>30 comments about 35 lines of code is a bit excessive..